### DISCUSSION TITLE
Codespaces don't just work out of the box

### DISCUSSION THREAD

**Discussion by Plasma** [community/community#38959] [Codespaces]
### Select Topic Area

Product Feedback

### Body

Hey team,

Like the concept of Codespaces, but twice I tried to just run/debug both a starter template and a single HTML file repository I had, but it doesn't launch and just open the browser/dev environment like I expected.

Would you be able to put together a simple example of the Codespace launching a hello-world plain HTML file, and a more exotic one like a C# project?

Thank you

**Comment by jfullstackdev** [comment]
what do you mean by that ? you want to code, say, a simple website right there and view the results, like a rendered web page ? 

for others, they need containers so all of the project will be pure online

for static sites, maybe just like this is your idea too 

https://srishti-56.medium.com/deploying-a-serverless-vue-js-app-using-github-codespaces-and-static-web-apps-5f5f51dabe76

**Comment by erikaellr** [comment]
Hi @Plasma, we're happy you're using Codespaces! 

Can you specify which Codespaces quick start template you're having a problem with, or provide a link to a public repo where the problem can be reproduced with step-by-step details about what actions you're taking and what you're expecting? 

I just tried the Flask and Express quick start templates, and both seemed to run/debug. The Express template starts the server automatically just after codespace startup with a simple browser on the right that I could pop-open in a new tab, and the Flask template allowed me to right click on or in `app.py` and run in terminal, which printed a localhost that when clicked, resolves to a github.dev URL for the running app. 

![express-quick-start](https://user-images.githubusercontent.com/1656972/201784869-3dfda6e8-8535-4116-8703-e194fef81946.png)
![flask-quick-start](https://user-images.githubusercontent.com/1656972/201784873-02a37aa6-66a5-4455-be83-84ba55589972.png)

**Comment by Plasma** [comment]
Thanks everyone for your responses; I've made a Bootstrap starter template repo at https://github.com/Plasma/codespace-test

I've then clicked on Code -> Codespaces and created one.

It loads in the web browser fine:
![image](https://user-images.githubusercontent.com/229209/202056868-e90c130c-6529-436f-ba36-9e5753e5c915.png)

Clicking on Run and Launch in chrome shows an error it can only work from VS Code Desktop:
![image](https://user-images.githubusercontent.com/229209/202056917-1cd743e9-2c3c-4418-affb-453d0300c799.png)

So I switch to VS Code desktop via the "Open in VS Code desktop" menu option, and try the same, but its only localhost that seems to be configured:

![image](https://user-images.githubusercontent.com/229209/202056999-d5c1b7a1-bcdb-48a4-9c4e-429e6cc99de4.png)

Do I need to configure something like devcontainer.json manually? If I can get this example working i'll then try a simple dotnet hello-world app next that needs compilation.

Thank you!

**Comment by joshaber** [comment]
You'll need something to serve the HTML & CSS. There are some extensions that can take care of that for you, for example https://marketplace.visualstudio.com/items?itemName=ritwickdey.LiveServer. And then once it's being served you can open the forwarded port to see it.
